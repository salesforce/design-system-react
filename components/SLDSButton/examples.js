// SAMPLE COMPONENT CODE -->
import React from 'react';

import SLDSButton from 'design-system-react/components/SLDSButton';

const ButtonExample = React.createClass({
	displayName: 'ButtonExample',

	getInitialState () {
		return {
			buttonStatefulSelected: false,
			buttonStatefulInverseSelected: false,
			buttonIconStateful: false
		};
	},

	render () {
		/* eslint-disable max-len */
		return (
			<div className="slds-grid slds-grid--vertical">

				<div className="slds-col | slds-m-bottom--small">
					<SLDSButton label="Base" />
					<SLDSButton label="Disabled" disabled />
				</div>
			</div>
		);
	},

	handleClickButtonStateful () {
		this.setState({ buttonStatefulSelected: !this.state.buttonStatefulSelected });
	},

	handleClickButtonStatefulInverse () {
		this.setState({ buttonStatefulInverseSelected: !this.state.buttonStatefulInverseSelected });
	},

	handleClickIconStateful () {
		this.setState({ buttonIconStateful: !this.state.buttonIconStateful });
	}
});

// <-- SAMPLE COMPONENT CODE

export default ButtonExample;
