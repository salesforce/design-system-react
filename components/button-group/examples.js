// SAMPLE COMPONENT CODE -->
import React from 'react';

import ButtonGroup from 'slds-for-react/button-group';
import Button from 'slds-for-react/button';
import Dropdown from 'slds-for-react/dropdown';
import Trigger from 'slds-for-react/dropdown/button-trigger';

const sampleData = [
	{
		id: 0,
		text: 'Menu Item One',
		value: '1'
	}, {
		id: 1,
		text: 'Menu Item Two',
		value: '2'
	}, {
		id: 2,
		text: 'Navigate to Google',
		value: '3',
		href: 'http://google.com'
	}
];

const ButtonGroupExample = React.createClass({
	displayName: 'ButtonGroupExample',

	getInitialState () {
		return {
			chartSelected: false,
			filterSelected: false
		};
	},

	render () {
		return (
			<div className="slds-grid slds-grid--vertical">
				<div className="slds-col | slds-m-bottom--medium">
					<ButtonGroup>
						<Button text="Refresh" theme="neutral" />
						<Button text="Edit" theme="neutral" />
						<Button text="Save" theme="neutral" />
						<Dropdown collection={sampleData} id="button-group-dropdown-example-1">
							<Trigger className="slds-button--last">
								<Button assistiveText="More Actions" iconStyle="icon-border" />
							</Trigger>
						</Dropdown>
					</ButtonGroup>
				</div>
				<div className="slds-col | slds-m-bottom--medium">
					<ButtonGroup>
						<Button
							iconCategory="utility"
							iconName="chart"
							iconStyle="icon-border"
							assistiveText="Chart"
							selectable
							selected={this.state.chartSelected}
							onClick={this.handleChartClick}
						/>
						<Button
							iconCategory="utility"
							iconName="filterList"
							iconStyle="icon-border"
							assistiveText="Filter"
							selectable
							selected={this.state.filterSelected}
							onClick={this.handleFilterClick}
						/>
						<Dropdown collection={sampleData} id="button-group-dropdown-example-2">
							<Trigger className="slds-button--last">
								<Button iconCategory="utility" iconName="sort" iconStyle="icon-more" assistiveText="More Actions" />
							</Trigger>
						</Dropdown>
					</ButtonGroup>
				</div>
			</div>
		);
	},

	handleChartClick () {
		this.setState({
			chartSelected: !this.state.chartSelected
		});
	},

	handleFilterClick () {
		this.setState({
			filterSelected: !this.state.filterSelected
		});
	}
});

// <-- SAMPLE COMPONENT CODE

export default ButtonGroupExample;
