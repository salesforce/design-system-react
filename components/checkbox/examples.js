// SAMPLE COMPONENT CODE -->
import React from 'react';

import Checkbox from 'slds-for-react/checkbox';

const CheckboxExample = React.createClass({
	displayName: 'CheckboxExample',

	getInitialState () {
		return {
			Apples: true,
			Oranges: false
		};
	},

	render () {
		return (
			<fieldset className="slds-form-element">
				<legend className="slds-form-element__label slds-form-element__label--top"></legend>
				<div className="slds-form-element__control">
					<Checkbox
						checked={this.state.Apples}
						labelText="Apples"
						onChanged={this._getChangeHandler('Apples')}
						value="Apples"
					/>
				</div>
				<div className="slds-form-element__control">
				<Checkbox
					checked={this.state.Oranges}
					labelText="Oranges"
					onChanged={this._getChangeHandler('Oranges')}
					value="Oranges"
				/>
				</div>
			</fieldset>
		);
	},

	_getChangeHandler (fruit) {
		return (checked) => {
			const newState = {};
			newState[fruit] = checked;
			this.setState(newState);
		};
	}
});

// <-- SAMPLE COMPONENT CODE

export default CheckboxExample;
