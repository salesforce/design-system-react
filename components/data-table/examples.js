// SAMPLE COMPONENT CODE -->
import React from 'react';
import sortBy from 'lodash/collection/sortBy';

import DataTable from 'slds-for-react/data-table';
import DataTableColumn from 'slds-for-react/data-table/column';
import DataTableRowActions from 'slds-for-react/data-table/row-actions';

const DataTableExample = React.createClass({
	displayName: 'DataTableExample',

	getInitialState () {
		return {
			collection: [
				{
					id: '8IKZHZZV80',
					name: 'Item One',
					count: 100976,
					lastModified: 'Yesterday',
					modifiedBy: 'Ashley McDougal'
				}, {
					id: '5GJOOOPWU7',
					name: 'Item Two',
					count: 54976,
					lastModified: 'Today',
					modifiedBy: 'Ashley McDougal'
				}, {
					id: 'Q8Z71ZUCEZ',
					name: 'Item Three',
					count: 10128,
					lastModified: 'Today',
					modifiedBy: 'Ashley McDougal'
				}, {
					id: 'WA0Q0XARAR',
					name: 'Item Four',
					count: 63616,
					lastModified: 'Yesterday',
					modifiedBy: 'Ashley McDougal'
				}
			],
			selection: []
		};
	},

	render () {
		return (
			<DataTable
				bordered
				collection={this.state.collection}
				columns={this.state.columns}
				id="DataTableExample-1"
				onChange={this.handleChanged}
				onSort={this.handleSort}
				selection={this.state.selection}
				selectRows
				striped
			>
				<DataTableColumn
					label="Campaign Name"
					property="name"
					truncate
				/>
				<DataTableColumn
					label="Count"
					property="count"
					sortable
					sortDirection={this.state.countSortDirection}
				/>
				<DataTableColumn
					label="Last Modified"
					property="lastModified"
					sortable
					sortDirection={this.state.lastModifiedSortDirection}
					truncate
				/>
				<DataTableRowActions
					collection={[
						{
							id: 0,
							text: 'Add to Group',
							value: '1'
						}, {
							id: 1,
							text: 'Publish',
							value: '2'
						}
					]}
					onAction={this.handleRowAction}
				/>
			</DataTable>
		);
	},

	handleChanged (selection) {
		this.setState({ selection });
	},

	handleRowAction (item, action) {
		console.log(item, action);
	},

	handleSort (sortColumn, sortDirection) {
		let rowData = this.state.collection;
		let countSortDirection;
		let lastModifiedSortDirection;

		if (sortColumn.property === 'count') {
			rowData = sortBy(rowData, 'count');
			countSortDirection = sortDirection;
		} else if (sortColumn.property === 'lastModified') {
			rowData = sortBy(rowData, 'lastModified');
			lastModifiedSortDirection = sortDirection;
		}

		if (sortDirection === 'desc') rowData.reverse();

		this.setState({
			collection: rowData,
			countSortDirection,
			lastModifiedSortDirection
		});
	}
});
// <-- SAMPLE COMPONENT CODE

export default DataTableExample;
