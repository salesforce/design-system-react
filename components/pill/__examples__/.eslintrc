{
  "rules": {
    "max-len": "off",
    "no-alert": "off",
    "no-console": "off",
    "no-unused-vars": "off",
	"prefer-template": "off"
  }
}
