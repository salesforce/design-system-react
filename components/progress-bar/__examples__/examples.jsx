import React from 'react';
import ProgressBar from '../../../components/progress-bar';

export const PROGRESS_0 = () => <ProgressBar id="progress-bar-1" value={0} />;
export const PROGRESS_25 = () => <ProgressBar id="progress-bar-2" value={25} />;
export const PROGRESS_HALF = () => (
	<ProgressBar id="progress-bar-3" value={50} />
);
export const PROGRESS_75 = () => <ProgressBar id="progress-bar-4" value={75} />;
export const COMPLETE_100 = () => (
	<ProgressBar id="progress-bar-5" value={100} color="success" />
);
export const COMPLETE_WITH_DESCRIPTION = () => (
	<ProgressBar value={100} color="success" label="Einstein Setup Assistant" />
);
export const ROUNDED_BAR = () => (
	<ProgressBar id="progress-bar-circular" value={20} radius="circular" />
);
export const X_SMALL_BAR = () => (
	<ProgressBar id="progress-bar-x-small" value={20} thickness="x-small" />
);
export const SMALL_BAR = () => (
	<ProgressBar id="progress-bar-small" value={20} thickness="small" />
);
export const MEDIUM_BAR = () => (
	<ProgressBar id="progress-bar-medium" value={20} thickness="medium" />
);
export const LARGE_BAR = () => (
	<ProgressBar id="progress-bar-large" value={20} thickness="large" />
);
