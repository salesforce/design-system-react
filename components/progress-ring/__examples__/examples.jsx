import React from 'react';
import ProgressRing from '../../../components/progress-ring';
import IconSettings from '../../icon-settings';
import Icon from '../../../components/icon';

class Example extends React.Component {
	static displayName = 'ProgressRingExamples';

	render() {
		return (
			<IconSettings iconPath="/assets/icons">
				<div className="slds-grid slds-grid_pull-padded slds-grid_vertical-align-center slds-wrap">
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={0} />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing
							value={0}
							hasIcon
							icon={<Icon category="utility" name="warning" />}
						/>
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing
							value={0}
							hasIcon
							icon={<Icon category="utility" name="lock" />}
						/>
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={20} theme="complete" />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={100} theme="complete" hasIcon />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={100} theme="complete" />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={20} theme="warning" />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={100} theme="warning" hasIcon />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={100} theme="warning" />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={20} theme="expired" />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={100} theme="expired" hasIcon />
					</div>
					<div className="slds-col_padded slds-m-top_x-large slds-size_1-of-3 slds-align_absolute-center">
						<ProgressRing value={100} theme="expired" />
					</div>
				</div>
			</IconSettings>
		);
	}
}

export default Example;
