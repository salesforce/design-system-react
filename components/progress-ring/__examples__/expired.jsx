import React from 'react';
import IconSettings from '~/components/icon-settings';
import ProgressRing from '~/components/progress-ring';

class Example extends React.Component {
	render () {
		return (
			<IconSettings iconPath="/assets/icons">
				<div className="slds-grid slds-grid--pull-padded slds-grid--vertical-align-center">
					<div className="slds-col--padded">
						<ProgressRing value={20} theme="expired" />
					</div>
					<div className="slds-col--padded">
						<ProgressRing value={20} theme="expired" hasIcon />
					</div>
					<div className="slds-col--padded">
						<ProgressRing value={100} theme="expired" />
					</div>
				</div>
			</IconSettings>
		);
	}
}

Example.displayName = 'ProgressRingDefault';

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
