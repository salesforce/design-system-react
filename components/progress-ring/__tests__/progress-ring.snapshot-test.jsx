/* eslint-env jest */
import { renderMarkup } from '../../../tests/snapshot-helpers';
import * as ProgressRing from '../__examples__/examples';

Object.keys(ProgressRing).forEach((name) => {
	test(`${name} DOM Snapshot`, () => {
		expect(renderMarkup(ProgressRing[name])).toMatchSnapshot();
	});
});
