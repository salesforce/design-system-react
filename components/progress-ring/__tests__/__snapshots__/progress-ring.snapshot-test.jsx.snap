// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default DOM Snapshot 1`] = `
"<div class=\\"slds-grid slds-grid_pull-padded slds-grid_vertical-align-center slds-wrap\\">
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"0\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 1 0 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"40\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 -0.8090169943749473 0.5877852522924732 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"0\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 1 0 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"><span><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg></span></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"0\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 1 0 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"><span><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#lock\\"></use></svg></span></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_complete\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_complete\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"40\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 -0.8090169943749473 0.5877852522924732 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_complete\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"><span title=\\"Complete\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#check\\"></use></svg></span></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_complete\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_warning\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_warning\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"40\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 -0.8090169943749473 0.5877852522924732 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_warning\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"><span title=\\"Warning\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg></span></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_warning\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_expired\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_expired\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"40\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 -0.8090169943749473 0.5877852522924732 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_expired\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"><span title=\\"Expired\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#error\\"></use></svg></span></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_expired\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_active-step\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_active-step\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"40\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 -0.8090169943749473 0.5877852522924732 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_active-step\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"70\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 -0.30901699437494756 -0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_active-step\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_active-step slds-progress-ring_large\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_warning slds-progress-ring_large\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"40\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 -0.8090169943749473 0.5877852522924732 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_expired slds-progress-ring_large\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"70\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" style=\\"transform:scaleX(1) rotate(-90deg)\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 -0.30901699437494756 -0.9510565162951535 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"><span title=\\"Expired\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#error\\"></use></svg></span></div>
    </div>
  </div>
  <div class=\\"slds-col_padded slds-m-top_x-large slds-size_1-of-4 slds-align_absolute-center\\">
    <div class=\\"slds-progress-ring slds-progress-ring_complete slds-progress-ring_large\\">
      <div aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\" class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
      <div class=\\"slds-progress-ring__content\\"></div>
    </div>
  </div>
</div>"
`;
